src/WorkflowCore/Services/ErrorHandlers/SuspendHandler.cs

```diff
@@ -21,7 +21,7 @@ public SuspendHandler(ILifeCycleEventPublisher eventPublisher, IDateTimeProvider
         public void Handle(WorkflowInstance workflow, WorkflowDefinition def, ExecutionPointer pointer, WorkflowStep step, Exception exception, Queue<ExecutionPointer> bubbleUpQueue)
         {
             workflow.Status = WorkflowStatus.Suspended;
-            _eventPublisher.PublishNotification(new WorkflowSuspended
+            _eventPublisher?.PublishNotification(new WorkflowSuspended
```

Comment:
The fact that the publisher can be disabled is leaking into where it is consumed, this creates inappropriate coupling.

Reply:
Why not just put the check in the publisher?